chain/actions/src/payload.rs

```diff
@@ -213,51 +224,155 @@ impl PayloadGenerator<TransactionRequest> for BasicPayloadGenerator {
-        }
-
-        let tx = TransactionRequest::default()
-            .with_input(closeIncomingChannelCall { source: source.into() }.abi_encode())
-            .with_to(self.contract_addrs.channels.into());
-        Ok(tx)
-    }
+    fn close_incoming_channel<S: IntoIterator<Item = Address>>(&self, sources: S) -> Result<TransactionRequest> {
+        let sources = sources.into_iter().collect::<Vec<_>>();
+        match sources.len() {
+            0 => Err(InvalidArguments("No incoming channel to close".into())),
+            1 => {
+                let source = sources.into_iter().next().unwrap();
+                if source.eq(&self.me) {
+                    return Err(InvalidArguments("Cannot close incoming channel from self".into()));
+                }
 
-    fn initiate_outgoing_channel_closure(&self, destination: Address) -> Result<TransactionRequest> {
-        if destination.eq(&self.me) {
-            return Err(InvalidArguments(
-                "Cannot initiate closure of incoming channel to self".into(),
-            ));
+                let tx = TransactionRequest::default()
+                    .with_input(closeIncomingChannelCall { source: source.into() }.abi_encode())
+                    .with_to(self.contract_addrs.channels.into());
+                Ok(tx)
+            }
+            _ => {
+                // create multisend data payload
+                let multisend_txns = sources
+                    .into_iter()
+                    .filter(|source| source != &self.me) // filter the incoming channel from self
+                    .map(|source| {
+                        let encoded_data = closeIncomingChannelCall { source: source.into() }.abi_encode();
+
+                        MultisendCallOnlyTransaction {
+                            encoded_data: encoded_data.into(),
+                            to: self.contract_addrs.channels.into(),
+                            value: U256::ZERO,
+                        }
+                    })
+                    .collect();
+
+                let tx_payload = MultisendCallOnlyTransaction::build_multisend_tx(multisend_txns);
+
+                let tx = TransactionRequest::default()
+                    .with_input(tx_payload)
+                    .with_to(MULTISEND_CALL_ONLY);
+                Ok(tx)
+            }
         }
+    }
 
-        let tx = TransactionRequest::default()
-            .with_input(
-                initiateOutgoingChannelClosureCall {
-                    destination: destination.into(),
+    fn initiate_outgoing_channel_closure<S: IntoIterator<Item = Address>>(
+        &self,
+        destinations: S,
+    ) -> Result<TransactionRequest> {
+        let destinations = destinations.into_iter().collect::<Vec<_>>();
+        match destinations.len() {
+            0 => Err(InvalidArguments("No outgoing channel to initiate closure".into())),
+            1 => {
+                let destination = destinations.into_iter().next().unwrap();
```

Comment:
Please, no `unwrap` or `expect` in code, when it's possible to express it more efficiently.